process_nwb/wavelet_transform.py

```diff
@@ -152,8 +164,71 @@ def wavelet_transform(X, rate, filters='rat', hg_only=True, X_fft_h=None, npad=N
     return Xh, X_fft_h, cfs, sds
 
 
+class ChBandIterator(AbstractDataChunkIterator):
+    def __init__(self, X, rate, filters='rat', npad=None, hg_only=True):
+        self.X = X
+        self.rate = rate
+        self.npad = npad
+
+        # Need to pad X before predicting chunk and filter shape:
+        X = self.X[:, 0]
+        npads, to_removes, _ = _npads(X, self.npad)
+        X = _smart_pad(X, npads)
+        self.ntimepts = X.shape[0]
+        self.filterbank, self.cfs, self.sds = get_filterbank(
+            filters, self.ntimepts, self.rate, hg_only
+        )
+
+        self.nch = self.X.shape[1]
+        self.nbands = len(self.filterbank)
+        self._i = 0
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        ch = self._i // self.nbands
+        band = self._i % self.nbands
+        self._i += 1
+
+        if ch >= self.X.shape[1]:
+            raise StopIteration
+
+        print("Doing channel {} band {}".format(ch, band))
+
+        if band == 0:
+            self.X_fft_h = None
+
+        X_ch = self.X[:, [ch]]
+        data, self.X_fft_h, cfs, sds = wavelet_transform(
+            X_ch,
+            self.rate,
+            filters=[self.filterbank[band]],
+            # X_fft_h=self.X_fft_h, # Reusing X_fft_h doesn't work here because wavelet_transform() decides whether to pad X based on whether X_fft_h is passed in. Could be optimized, but this is a small price to pay for the memory gains
```

Comment:
I'd just remove this and create a github issue to investigate whether this trick can be used.

Reply:
> I added some utils to helping testing these type of functions
> 
> https://github.com/BouchardLab/process_nwb/blob/acf598b650bf6d2d67980d4f399c7d5ce19177c6/process_nwb/utils.py#L139
> 
> 
> https://github.com/BouchardLab/process_nwb/blob/acf598b650bf6d2d67980d4f399c7d5ce19177c6/process_nwb/utils.py#L191
> 
> Example test that uses them here (you could create a second test that does does the wavelet step, or add another set of asserts in the existing test with this flag used.
> https://github.com/BouchardLab/process_nwb/blob/master/tests/test_pipeline.py

Thanks, will do.